ext/bson/write.c

```diff
@@ -383,6 +384,35 @@ void pvt_put_int32(byte_buffer_t *b, const int32_t i)
   b->write_position += 4;
 }
 
+/* The docstring is in init.c. */
+VALUE rb_bson_byte_buffer_put_uint32(VALUE self, VALUE i)
+{
+  byte_buffer_t *b;
+
+  if (RB_TYPE_P(i, T_FLOAT)) {
+    rb_raise(rb_eArgError, "put_uint32; incorrect type: float, expected: integer");
+  }
+
+  int64_t temp = NUM2LL(i);
```

Comment:
All of the declarations should be at the top of the scope for compatibility, in this case at the beginning of the function.